frontend/app/utils/string-utils.ts

```diff
@@ -37,3 +39,97 @@ export function padWithZero(value: number, maxLength: number): string {
 export function trimToUndefined(str: string | undefined): string | undefined {
   return Number(str?.trim().length) > 0 ? str : undefined;
 }
+
+/**
+ * Arguments for formatting just the address line
+ */
+interface FormatAddressLineArguments {
+  /** Street address */
+  address: string;
+  /** Apartment, suite, or unit number */
+  apartment?: string;
+}
+
+/**
+ * Formats an apartment/suite number with the address
+ * @param params Address line formatting parameters
+ * @returns Formatted address line
+ */
+export function formatAddressLine({ address, apartment }: FormatAddressLineArguments): string {
+  if (!apartment?.trim()) {
+    return address;
+  }
+
+  // Check if apartment is a simple alphanumeric suite number
+  const isSuiteNumber = /^[a-z\d]+$/i.test(apartment.trim());
+
+  return isSuiteNumber ? `${apartment.trim()}-${address}` : `${address} ${apartment.trim()}`;
+}
+
+/**
+ * Arguments for formatting a complete address
+ */
+export interface FormatAddressArguments {
+  /** Street address */
+  address?: string;
+
+  /** City name */
+  city: string;
+
+  /** Country name */
+  country: string;
+
+  /** Province or state code (optional) */
+  provinceState?: string;
+
+  /** Postal or ZIP code (optional) */
+  postalZipCode?: string;
+
+  /** Apartment, suite, or unit number (optional) */
+  apartment?: string;
+
+  /**
+   * The format of the address
+   *
+   * - `standard`: The standard address format, with the address line, city, province/state, postal/zip code, and country.
+   * - `alternative`: An alternative address format, with the address line, city, province/state, postal/zip code, and country on separate lines.
+   */
+  format?: 'standard' | 'alternative';
+}
+
+/**
+ * Formats an address string based on the provided arguments.
+ *
+ * @param params Address formatting parameters
+ * @returns Formatted address string
+ */
+export function formatAddress({
```

Comment:
I think it would be beneficial to add an example output for each format, here. It takes a lot of mental strain (for me, at least) to read the code and try to infer what the output will look like.